Parser/Tree_Nodes/r_asmt.c

```diff
@@ -24,27 +24,45 @@ r_asmt *create_r_asmt(GArray *tokenStream, unsigned long index, unsigned long *n
 
     r_asmt *new_asmt = calloc(1, sizeof(r_asmt));
 
-    // Need to try expression before we add the ID, because it cannot be self-referencing
-    new_asmt->expression = create_expr(tokenStream, index + 2, next);
-
-    Type *varType = malloc(sizeof(Type));
-
+    //Get expression so we know the type
     // Get the variable in the reassignment
-    Token *curToken = &g_array_index(tokenStream, Token, index);
-
+    Token *curToken = &g_array_index(tokenStream, Token, (*next));
```

Comment:
You should be using index, not next. Next is really only used after you create something with an unknown length. For example, when creating an assigmnment, we create the expression, then use next to make sure the following token is a semicolon